base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class BaseDecorator < Nameable
```

Comment:
Consider making `nameable` accessible for read and write since you do not have any manual getter and setter for this instance variable. You can do that by having the `attr_accessor` property for the `nameable`.